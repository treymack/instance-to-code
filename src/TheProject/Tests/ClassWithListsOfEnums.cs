using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class ClassWithListsOfEnums
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new EnumListClass
            {
                EnumList1 = new List<Enum1>
                {
                    { Enum1.Value1 },
                    { Enum1.Value1 },
                },
                EnumList2 = new List<Enum2>
                {
                    { Enum2.Value2 },
                    { Enum2.Value1 },
                },
                EmptyList = new List<Enum1> { },
            };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }
  
    public class EnumListClass
    {
        public List<Enum1> EnumList1 { get; set; }
        public List<Enum2> EnumList2 { get; set; }
        public List<Enum1> EmptyList { get; set; }
    }
}
