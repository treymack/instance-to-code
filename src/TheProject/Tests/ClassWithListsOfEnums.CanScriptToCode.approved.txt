new TheProject.Tests.EnumListClass()
{
EmptyList = new System.Collections.Generic.List<TheProject.Tests.Enum1>
{},
EnumList1 = new System.Collections.Generic.List<TheProject.Tests.Enum1>
{TheProject.Tests.Enum1.Value1, 
TheProject.Tests.Enum1.Value1, 
},
EnumList2 = new System.Collections.Generic.List<TheProject.Tests.Enum2>
{TheProject.Tests.Enum2.Value2, 
TheProject.Tests.Enum2.Value1, 
},
}